crates/task-impls/src/consensus.rs

```diff
@@ -453,24 +458,42 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>, A: ConsensusApi<TYPES, I> +
                     return;
                 }
 
-                // Verify a timeout certificate exists and is valid
+                // Verify a timeout certificate OR a view sync certificate exists and is valid.
                 if proposal.data.justify_qc.get_view_number() != view - 1 {
```

Comment:
Please advise if this is okay. We explicitly first check if there exists a timeout cert, THEN we check if there is a view_sync cert. But the checks are independent. If we have neither, then we exit as normal.

Reply:
Yes this is ok.  We should codify this by making the propsal have and `Option<Either<TimeoutCert, ViewSyncFinalizedCert>>` instead of two options

Reply:
This can be a follow up task like we discussed in the team call